### PR TITLE
Exceptions - provide utests

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] covers exceptions missing tests

### CODE REVIEW

File: tests/ConfigurationException/InvalidConfigurationExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public function testDefaults()
+    {
+        $exception = new ConfigurationException\InvalidFixerConfigurationException(
+            'hal',
+            'I cannot do that, Dave.'
+        );
+
+        $this->assertSame(Console\Command\FixCommand::EXIT_STATUS_FLAG_HAS_INVALID_FIXER_CONFIG, $exception->getCode());
```
Comment (julienfalque) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: I would add an assertion on `getMessage()` result here on other `testDefaults`.

File: tests/ConfigurationException/InvalidConfigurationExceptionTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\ConfigurationException;
+
+use PhpCsFixer\ConfigurationException;
```
Comment (localheinz) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: Personally I prefer to limit the number of imports, but adjusted now!

**Comment by localheinz** [comment]
Thank you, @julienfalque!

**Comment by localheinz** [comment]
@julienfalque 

Added a few more tests to cover the yet uncovered exceptions!

File: tests/Linter/LintingExceptionTest.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+use PHPUnit\Framework;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\Command\DescribeNameNotFoundException
+ */
+final class LintingExceptionTest extends Framework\TestCase
```
Comment (julienfalque) [inline tests/Linter/LintingExceptionTest.php]: Please add a `testConstructorSetsValues` test here and in `UnavailableLinterExceptionTest`: those exceptions are not marked as `@internal` so the tests should guaranty that the constructor does not break (even if it is the inherited constructor).

File: tests/Linter/LintingExceptionTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+use PhpCsFixer\Linter\LintingException;
+use PHPUnit\Framework;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\Command\DescribeNameNotFoundException
```
Comment (julienfalque) [inline tests/Linter/LintingExceptionTest.php]: The covered class is wrong (same in `UnavailableLinterExceptionTest`).

File: tests/ConfigurationException/InvalidConfigurationExceptionTest.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+use PHPUnit\Framework;
+
+/**
+ * @author Andreas Möller <am@localheinz.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException
+ */
+final class InvalidConfigurationExceptionTest extends Framework\TestCase
```
Comment (SpacePossum) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: will this work on the old PHPUnit versions?

Comment (localheinz) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: Has been backported!

**Comment by localheinz** [comment]
@julienfalque 

Sorted out!

File: tests/ConfigurationException/InvalidConfigurationExceptionTest.php (hunk 5 above)
Comment (keradus) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: @SpacePossum except of this particular case...
if not, our Travis would crash on lowest deps ;)

Comment (localheinz) [inline tests/ConfigurationException/InvalidConfigurationExceptionTest.php]: 👍

**Comment by SpacePossum** [comment]
@localheinz please remove the not used import;
https://travis-ci.org/FriendsOfPHP/PHP-CS-Fixer/jobs/240741611#L880

and this should be good to go :)

**Comment by localheinz** [comment]
Thank you, @julienfalque, @keradus, and @SpacePossum!

File: tests/ConfigurationException/RequiredFixerConfigurationExceptionTest.php [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+        $this->assertSame(sprintf('[%s] %s', $fixerName, $message), $exception->getMessage());
+        $this->assertSame(FixCommand::EXIT_STATUS_FLAG_HAS_INVALID_FIXER_CONFIG, $exception->getCode());
+        $this->assertNull($exception->getPrevious());
+    }
+
+    public function testConstructorSetsValues()
+    {
+        $fixerName = 'hal';
+        $message = 'I cannot do that, Dave.';
```
Comment (keradus) [inline tests/ConfigurationException/RequiredFixerConfigurationExceptionTest.php]: :D

Comment (localheinz) [inline tests/ConfigurationException/RequiredFixerConfigurationExceptionTest.php]: maxresdefault

**Comment by localheinz** [comment]
Thank you, @julienfalque, @keradus, and @SpacePossum!

**Comment by keradus** [comment]
Thank you @localheinz.

**Pull request closed without merging**